### DISCUSSION TITLE
Games played limits?

### DISCUSSION THREAD

**Discussion by jaredmontierth** [cwendt94/espn-api#501] [Q&A]
Is there a games played attribute for basketball? I did some digging but wasn't able to find it or anything similar. It would be useful to have since the number of games allowed changes week to week (depending on how many NBA games there are in that week) and I want to make a tool that uses that info.

**Comment by cwendt94** [comment]
Are you thinking about per player the games they played that week, or total games they have played in the season?

**Reply by jaredmontierth** [reply]
Yeah so there's a setting in basketball leagues to set a limit on games played by fantasy teams per week. In my league I have it set to 36. The limit changes depends on if there are days off (so this week the limit is 31 since there are no games this Sunday). 
After each day the number is updated based on how many players a team played that day, and I'd love to be able to access that number with this API. 

<img width="657" alt="games played limit option in league settings" src="https://github.com/cwendt94/espn-api/assets/78840493/c7f5c449-7ddd-46db-8759-63983de82961">



<img width="378" alt="what it looks like in a matchup. currently at 19 games played out of 31 for the week" src="https://github.com/cwendt94/espn-api/assets/78840493/0aacd3e5-4357-4cdc-949a-5d1e9a4b048e">

Thanks for maintaining this awesome API!

**Reply by cwendt94** [reply]
Oh okay, I have never seen that before. If ESPN shows it, I should be able to get it from their API. What view is this data shown on? Like your team view, matchup, box score?

**Reply by jaredmontierth** [reply]
It shows up at the top of the box score page, which is where I got that second screenshot. The url is like https://fantasy.espn.com/basketball/boxscore?leagueId={}&matchupPeriodId={}&seasonId={}&teamId={}

**Reply by cwendt94** [reply]
Perfect, I will check out the API data from ESPN!

**Reply by jaredmontierth** [reply]
I did a debug call on my league and found this:
`"statBySlot": {"15": {"exceededOnScoringPeriod": 13, "limitExceeded": true, "statId": 42, "value": 40.0}}, "ties": 0, "wins": 0}, "gamesPlayed": 0, "pointsByScoringPeriod": {"7": 238.0, "8": 114.0, "9": 291.0, "10": 75.0, "11": 180.0, "12": 240.0, "13": 239.0}, "teamId": 5, "totalPoints": 1377.0}` 

`value` seems to be the amount of games a team has used -- this was a case where a team had 40/36 (it is pretty standard and good strategy to exceed the limit on the last day, since it can't penalize you the next day -- just a weird ESPN league quirk, some discussion about it [here](https://www.reddit.com/r/fantasybball/comments/3r8kkj/kinda_confused_how_the_game_limits_work_on_espn/))

<img width="700" alt="Screenshot 2023-12-06 at 4 32 07 PM" title="example of exceeding GP limit -- pointsByScoringPeriod should line up" src="https://github.com/cwendt94/espn-api/assets/78840493/e409f82e-851a-4db2-9627-5c45bcfaac05">


Since this team maxed out, I think that sets `limitExceeded` to true and assigns a value for `exceededOnScoringPeriod` which I think is the day that the team reached the max. `statId` being `42` maps to `GP` in STATS_MAP in constant.py which makes sense. 

The only thing I wasn't able to find was where the weekly limit lives (the 36 in the screenshot from this example). Control F is only getting me so far, lol. Hopefully this helps maybe a little.

**Reply by cwendt94** [reply]
Thats definitely helpful! Im still trying to figure out where that value is sourced as well. Would you be able to capture the API call that ESPN is making when you go to Box Score on their website. If you inspect the page and go to the network tab you should see an API call that looks like this `https://lm-api-reads.fantasy.espn.com/apis/v3/games/fba/seasons/2024/segments/0/leagues/1301764861?scoringPeriodId=6&view=mBoxscore&view=mMatchupScore&view=mRoster&view=mSettings&view=mStatus&view=mTeam&view=modular&view=mNav`.  You should be able to see the data under Preview and Response tab. This could help trying to figure out where they are storing those values.
